### PR TITLE
pioasm: init at 1.5.1

### PR DESCRIPTION
**Pull request by emilytrau**
## Description of changes

Assemble PIO programs for Raspberry Pi Pico https://github.com/raspberrypi/pico-sdk

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/pi/pioasm/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Assemble PIO programs for Raspberry Pi Pico";
+    homepage = "https://github.com/raspberrypi/pico-sdk";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ emilytrau ];
+    platforms = platforms.unix;
```
Comment (MikaelFangel) [inline pkgs/by-name/pi/pioasm/package.nix:35]:
```suggestion
    platforms = platforms.unix;
    mainProgram = "pioasm";
```

Comment (emilytrau) [reply pkgs/by-name/pi/pioasm/package.nix:35]: This should already be default since it's the same as the package name?

Comment (MikaelFangel) [reply pkgs/by-name/pi/pioasm/package.nix:35]: Can't remember where it read it, but you can see from the comment on line 149 that it is deprecated.

[https://github.com/NixOS/nixpkgs/blob/master/lib/meta.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/meta.nix)

Let me know if I need to provide more details. (I'm on my phone, hence the short answer)

Comment (emilytrau) [reply pkgs/by-name/pi/pioasm/package.nix:35]: ooo i learned something new thanks!

**Review by MikaelFangel — APPROVED** [review]
Result of nixpkgs-review on linux x86-64:
<details>
<summary>1 package built:</summary>
pioasm
</details>

**Pull request merged**